src/util/terminationPreprocessing.ml

```diff
@@ -0,0 +1,65 @@
+open GoblintCil
+
+module VarToStmt = Map.Make(CilType.Varinfo) (* maps varinfos (= loop counter variable) to the statement of the corresponding loop*)
+
+let extract_file_name s =                    (*There still may be a need to filter more chars*)
+  let ls = String.split_on_char '/' s in    (*Assuming '/' as path seperator*)
+  let ls = List.rev ls in
+  let s' = List.nth ls 0 in
+  let ls = String.split_on_char '.' s' in
+  let s' = List.nth ls 0 in
+  let without_spaces = String.split_on_char ' ' s' in
+  let s' = String.concat "" without_spaces in
+  s'
```

Comment:
Why all this? Besides reimplementing standard path-manipulation functions, this is also error-prone. Basenames are not always unique.

Reply:
I simplified it now by keeping the line numbers plus a unique counter within this function.